source/javascripts/lib/wirsing/filters/switch.coffee

```diff
@@ -0,0 +1,17 @@
+@wirsing ?= {}
+wirsing.filter ?= {}
+
+wirsing.filter.switch =
+  on: (items, key) ->
+    switchObject = {}
+
+    switches = ->
+      if Object.keys(switchObject).length == 0
```

Comment:
premature optimization is the root of all evil.
I don't think we need to lazy-initialize the `switchObject` which makes all other lines of code much more complicated.
I also don't like calling things "Object" when it's pretty clear that it is an object :-D
And `keys` might be better than `names` to retrieve all keys :-D

How about this:
```CoffeeScript
on: (items, key) ->
  switches = {}
  switches[item[key]] ?= true for item in items
  
  return {
    keys: -> Object.keys switches
    allShown: -> switches
    apply: (isShown) -> item for item in items when isShown[item[key]]
  }
```